Mathlib/Combinatorics/HalesJewett.lean

```diff
@@ -88,16 +180,61 @@ structure Line (α ι : Type*) where
 #align combinatorics.line Combinatorics.Line
 
 namespace Line
+variable {l : Line α ι} {i : ι} {a x : α}
+
+/-- Consider a line `l : Line α ι` as a function `α → ι → α`. -/
+@[coe] def toFun (l : Line α ι) (x : α) (i : ι) : α := (l.idxFun i).getD x
 
 -- This lets us treat a line `l : Line α ι` as a function `α → ι → α`.
-instance (α ι) : CoeFun (Line α ι) fun _ => α → ι → α :=
+instance instCoeFun : CoeFun (Line α ι) fun _ => α → ι → α :=
   ⟨fun l x i => (l.idxFun i).getD x⟩
 
+lemma coe_apply (l : Line α ι) (x : α) (i : ι) : l x i = (l.idxFun i).getD x := rfl
+
+-- Note: This is not made a `FunLike` instance to avoid having two syntactically different coercions
```

Comment:
Can you explicit these two possible coercions in the comment?

Reply:
What do you mean? The two coercions are defeq, but one is wrapped in `FunLike.coe`.

Reply:
I didn't understand what you meant exactly. 
Just write what these coercions are, mathematically, or syntactically.

Reply:
Better now?